lib/exec.ts

```diff
@@ -162,7 +165,9 @@ export function executeDirect(
                 truncated: streams.truncated,
                 execTime: ((endTime - startTime) / BigInt(1000000)).toString(),
             };
-            logger.debug('Execution', {type: 'executed', command: command, args: args, result: result});
+            // It's not safe to log the entire `result` here, as one or more
+            // streams may be at the string limit.
+            logger.debug('Execution', {type: 'executed', command: command, args: args});
```

Comment:
I'm not a big fan of this change to not show the result. Although sometimes noisy, it is very helpful during development and debugging of compilers and tools.

Reply:
Ah okay, as an alternative then, perhaps we can just wrap this in a test for whether the debug level is actually enabled (so the potentially huge string is never prepared if the log level is disabled)...?

Reply:
I have restored result logging by checking the logging level in the latest version.

Reply:
thanks